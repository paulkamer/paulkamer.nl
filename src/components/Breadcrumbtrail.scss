@import '../styles/variables';
@import '../styles/mixins';

.breadcrumbtrail {
  @include mobile {
    padding-top: 1em;
  }

  ul {
    padding: 0;
    list-style: none;

    li {
      display: inline;
    }

    li:not(:last-of-type)::after {
      padding: 0 0.25em;
      content: "\203A";
    }
  }
}
