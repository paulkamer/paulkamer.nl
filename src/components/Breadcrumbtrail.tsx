import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import './Breadcrumbtrail.scss';

type BreadCrumb = {
  label: string;
  to?: string;
}

export interface BreadcrumbtrailProps { trail: BreadCrumb[]; }

export class Breadcrumbtrail extends Component<BreadcrumbtrailProps> {
  render() {
    return (
      <nav className="breadcrumbtrail">
        <ul>
          <li><Link to="/">Home</Link></li>
          {
            this.props.trail.map((crumb, index) =>
              <li key={`breadcrumb_${index}`}>{crumb.to ? <Link to={crumb.to}>{crumb.label}</Link> : crumb.label}</li>
            )
          }
        </ul>
      </nav>
    );
  }
}
