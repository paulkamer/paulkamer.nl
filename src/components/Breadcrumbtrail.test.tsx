import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { BreadCrumbtrail } from './Breadcrumbtrail';

const testTrailItems = [
  {
    label: 'Jobs',
    to: '/jobs'
  },
  {
    label: 'Non-link item'
  }
];

it('renders correctly without any trail items', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <BreadCrumbtrail trail={[]} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with trail items', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <BreadCrumbtrail trail={testTrailItems} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
