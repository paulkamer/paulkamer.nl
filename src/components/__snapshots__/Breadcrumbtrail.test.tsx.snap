// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with trail items 1`] = `
<nav
  className="breadcrumbtrail"
>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/jobs"
        onClick={[Function]}
      >
        Jobs
      </a>
    </li>
    <li>
      Non-link item
    </li>
  </ul>
</nav>
`;

exports[`renders correctly without any trail items 1`] = `
<nav
  className="breadcrumbtrail"
>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
  </ul>
</nav>
`;
